src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/ContractFrame/index.tsx

```diff
@@ -0,0 +1,253 @@
+    id: 'components.DashboardItem.Order.ContractFrame.errorSubmitForSignature',
+  },
+  errorMaxPolling: {
+    defaultMessage: 'The signature is taking more time than expected ... please come back later.',
+    description:
+      'Message displayed inside the contract signin modal if the order polling has reached its maximum attempts.',
+    id: 'components.DashboardItem.Order.ContractFrame.errorMaxPolling',
+  },
+  errorPolling: {
+    defaultMessage: 'An error happened while fetching the order. Please come back later.',
+    description:
+      'Message displayed inside the contract signin modal if the order polling failed a request.',
+    id: 'components.DashboardItem.Order.ContractFrame.errorPolling',
+  },
+  loadingContract: {
+    defaultMessage: 'Loading your contract ...',
+    description: 'Message displayed inside the contract signin modal when loading the contract.',
+    id: 'components.DashboardItem.Order.ContractFrame.loadingContract',
+  },
+  polling: {
+    defaultMessage: 'Verifying signature ...',
+    description: 'Message displayed inside the contract signin modal when polling the order.',
+    id: 'components.DashboardItem.Order.ContractFrame.polling',
+  },
+  finishedCaption: {
+    defaultMessage: 'Congratulations!',
+    description: 'Title displayed inside the contract signin modal when the contract is signed.',
+    id: 'components.DashboardItem.Order.ContractFrame.finishedCaption',
+  },
+  finishedDescription: {
+    defaultMessage:
+      'You will receive an email containing your contract signed. You can now enroll to your course runs!',
+    description: 'Message displayed inside the contract signin modal when the contract is signed.',
+    id: 'components.DashboardItem.Order.ContractFrame.finishedDescription',
+  },
+  finishedButton: {
+    defaultMessage: 'Next',
+    description: 'Button displayed inside the contract signin modal when the contract is signed.',
+    id: 'components.DashboardItem.Order.ContractFrame.finishedButton',
+  },
+});
+
+interface Props {
+  order: CredentialOrder;
+  product: Product;
+  isOpen: boolean;
+  onDone: () => void;
+  onClose: () => void;
+}
+
+const DUMMY_REGEX = /https:\/\/dummysignaturebackend.fr/;
+
+enum SignatureType {
+  LEX = 'lex',
+  DUMMY = 'dummy',
+}
+
+enum ContractSteps {
+  LOADING_CONTRACT,
+  SIGNING,
+  POLLING,
+  FINISHED,
+  ERROR,
+  NONE,
+}
+
+export interface SignatureProps {
+  onDone: () => void;
+  onError: (error: string) => void;
+  invitationLink: string;
+}
+
+export const ContractFrame = (props: Props) => {
+  return (
+    <Modal
+      isOpen={props.isOpen}
+      shouldCloseOnOverlayClick={false}
+      shouldCloseOnEsc={false}
+      onRequestClose={props.onClose}
+      testId={'contract-frame__' + props.order.id}
+    >
+      {/* We want to have a fresh component when we close and re-open the modal to avoid any */}
+      {/* state resetting issues. */}
+      {props.isOpen && <ContractFrameContent {...props} />}
```

Comment:
I wonder if the Modal does not manage that internally ?

Reply:
Well yes, you're right!